contracts/token/ERC20/ERC20.sol

```diff
@@ -171,7 +171,8 @@ contract ERC20 is IERC20 {
     function _approve(address owner, address spender, uint256 value) internal {
         require(owner != address(0), "ERC20: approve from the zero address");
         require(spender != address(0), "ERC20: approve to the zero address");
-
+        require(spender != msg.sender, "ERC20: approval to current owner");
```

Comment:
If there is necessary, maybe we can change the comments: "ERC20: approve to your own address", or something better.